### PR TITLE
upgrade shipped suitesparse components to 7.8.0

### LINKED ISSUE #37192: update  SuiteSparse to v.7.6, allow its headers to be in suitesparse/ subdir
### Problem Description

SuiteSparse is now on GitHub: https://github.com/DrTimothyAldenDavis/SuiteSparse and has a new release, 7.6.0. We are still on major version 5. v.7.6.0 is already provided by Homebrew, see #37149 for the related issue.

The SuiteSparse headers have been moved to `suitesparse/*` - which of course makes perfect sense, but needs to support older versions adds a complication (not unknown to us though)

SuiteSparse now builds with cmake, which is nice.

### Proposed Solution

We should upgrade, and allow headers to be in the right place. 

Cf as well #37161 and 

### Is there an existing issue for this?

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.

### PR DESCRIPTION
**Pull request by kiwifb**
This is to fix #37192 and supersedes #34206 
The new cmake superbuild is leveraged to only build the suitesparse components of interest
* amd
* camd
* colamd
* ccolamd
* cholmod
* umfpack
 
### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by dimpase — APPROVED** [review]
lgtm

**Comment by dimpase** [comment]
please feel free to adjust tags accordingly

**Comment by dimpase** [comment]
the failing doctest
```
sage -t --long --random-seed=286735480429121101562228604801325644303 sage_setup/clean.py  # 1 doctest failed
```
looks irrelevant to this PR

**Comment by kiwifb** [comment]
Sorry, I have been busy. I think I need to explain some of the options I set with cmake a little bit more before shipping it.

**Comment by kiwifb** [comment]
Since all I have done in the last commit is add some text, I will label it positive review as approved by Dima.

File: build/pkgs/suitesparse/SPKG.rst [hunk 1]
```diff
@@ -2,26 +2,19 @@ suitesparse: A suite of sparse matrix software
 ==============================================
 
 SuiteSparse is a collection of software to deal with sparse matrix. It is
-hosted at http://faculty.cse.tamu.edu/davis/suitesparse.html
+hosted at https://people.engr.tamu.edu/davis/suitesparse.html with source code
+now on github at https://github.com/DrTimothyAldenDavis/SuiteSparse
 
-This spkg does a minimal install of suitesparse disabling the following
+This spkg does a minimal install of suitesparse only intalling the following
```
Comment (mkoeppe) [inline build/pkgs/suitesparse/SPKG.rst]:
```suggestion
This spkg does a minimal install of suitesparse, only installing the following
```

**Comment by mkoeppe** [comment]
Has this been tested on more than one machine? 

"CI Linux Incremental" is only being run now that the label "c: packages: standard" is set.
(See #37222 for that.)

**Comment by kiwifb** [comment]
Well, some CI ran before, but yes, I do not think it ran on Mac OS before for example. I'll just push your comma change and remove the positive review label until the full CI ran.

**Comment by mkoeppe** [comment]
I'm running the full CI Linux in https://github.com/mkoeppe/sage/actions/runs/7748353404 and CI macOS in https://github.com/mkoeppe/sage/actions/runs/7748353409

**Comment by mkoeppe** [comment]
Various failures:
"CMake 3.22 or higher is required.  You are running version 3.20.2" on https://github.com/mkoeppe/sage/actions/runs/7748353404/job/21130825606#step:14:490 ([standard-pre / linux (almalinux-9-python3.11, standard)](https://github.com/mkoeppe/sage/actions/runs/7748353404/job/21130825606#logs))

```
In file included from /sage/local/var/tmp/sage/build/suitesparse-7.6.0/src/CHOLMOD/Include/cholmod.h:308,
                   from /sage/local/var/tmp/sage/build/suitesparse-7.6.0/src/CHOLMOD/Include/cholmod_internal.h:33,
                   from /sage/local/var/tmp/sage/build/suitesparse-7.6.0/src/CHOLMOD/Check/cholmod_check.c:65:
  /sage/local/var/tmp/sage/build/suitesparse-7.6.0/src/SuiteSparse_config/SuiteSparse_config.h:583:65: error: expected ')' before ',' token
    583 |     #define SUITESPARSE_BLAS(name,NAME) SUITESPARSE_FORTRAN(name,NAME)
        |                                                                 ^
```
(https://github.com/mkoeppe/sage/actions/runs/7748353404/job/21130813895#step:14:1077, [minimal-pre / linux (ubuntu-lunar, minimal)](https://github.com/mkoeppe/sage/actions/runs/7748353404/job/21130813895#logs))

Likewise on many other platforms. This sounds familiar from the last attempt. I'll take a closer look in the next days.

**Comment by mkoeppe** [comment]
The cmake version tightening was already on the old branch, can just cherry pick from there - https://github.com/sagemath/sagetrac-mirror/commit/6f4032e5d6392028e540fe67f9a91a75e5e93b4c

**Comment by kiwifb** [comment]
Sounds like something that's overdue in any case. I will deal with it ASAP.

**Comment by kiwifb** [comment]
There are three build failure of suitesparse 7.6.0 so far but I do not have enough of the build log to make a diagnostic.

**Comment by mkoeppe** [comment]
You can see the full build log in the section "Print out logs for immediate inspection", e.g. https://github.com/sagemath/sage/actions/runs/7752508002/job/21142181161?pr=37204, clock on the triangle to expand the collapsed portion

**Comment by kiwifb** [comment]
Thanks, found it. There is mismatch between system gcc (13) and sage's gfortran (12). I am suspecting that's the issue. I think turning off fortran should be OK, so I'll try that.

**Comment by mkoeppe** [comment]
IIRC, that was also the issue last time. It may be worth trying if just disabling LTO would solve this.

**Comment by kiwifb** [comment]
What Flag would that be?

**Comment by mkoeppe** [comment]
`-fno-lto`, I think; untested

**Comment by kiwifb** [comment]
`-fno-lto` (which should be correct according to documentation had no effect. Let's see if removing fortran support has an effect.

**Comment by kiwifb** [comment]
The last conda tests appear to be stuck. The failing tests so far look unrelated.

**Comment by kiwifb** [comment]
OK, retrying here, will move 7.8.0 (7.8.1 is identical apart from graphblas which we do not use).

**Comment by kiwifb** [comment]
This PR is now about suitesparse 7.8.0

**Comment by mkoeppe** [comment]
Portability tests (with a bunch of other upgrades) at https://github.com/mkoeppe/sage/actions/runs/10484245959

**Comment by mkoeppe** [comment]
Builds and tests OK on macOS (x86_64) with homebrew

**Comment by mkoeppe** [comment]
Looking good, thanks for preparing this upgrade!

**Comment by mkoeppe** [comment]
@vbraun 
```
  [suitesparse-7.8.0] https://ftp.sun.ac.za/ftp/pub/mirrors/www.sagemath.org/spkg/upstream/suitesparse/SuiteSparse-7.8.0.tar.gz
  [suitesparse-7.8.0] [xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx]
  [suitesparse-7.8.0] ERROR [transfer|run:135]: [Errno socket error] [Errno 404] Not Found: '//ftp.sun.ac.za/ftp/pub/mirrors/www.sagemath.org/spkg/upstream/suitesparse/SuiteSparse-7.8.0.tar.gz'
  [suitesparse-7.8.0] Traceback (most recent call last):
  [suitesparse-7.8.0]   File "/home/runner/work/sage/sage/build/bin/sage-package", line 40, in <module> [....]
  [suitesparse-7.8.0] sage_bootstrap.tarball.FileNotMirroredError: tarball does not exist on mirror network
```
https://github.com/sagemath/sage/actions/runs/10691572603/job/29638332632#step:4:3332

**Pull request merged**